code/modules/industrial_lift/lift_master.dm

```diff
@@ -0,0 +1,395 @@
+	if(isatom(atom_reference_OR_z_level_number))
+		z = atom_reference_OR_z_level_number.z
+
+	if(!isnum(z) || z < 0 || z > world.maxz)
+		return null
+
+	var/list/platforms_in_z = list()
+
+	for(var/obj/structure/industrial_lift/lift_to_check as anything in lift_platforms)
+		if(lift_to_check.z)
+			platforms_in_z += lift_to_check
+
+	return platforms_in_z
+
+/**
+ * Moves the lift UP or DOWN, this is what users invoke with their hand.
+ * This is a SAFE proc, ensuring every part of the lift moves SANELY.
+ * It also locks controls for the (miniscule) duration of the movement, so the elevator cannot be broken by spamming.
+ * Arguments:
+ * going - UP or DOWN directions, where the lift should go. Keep in mind by this point checks of whether it should go up or down have already been done.
+ * user - Whomever made the lift movement.
+ */
+/datum/lift_master/proc/MoveLift(going, mob/user)
+	set_controls(LIFT_PLATFORM_LOCKED)
+	//lift_platforms are sorted in order of lowest z to highest z, so going upwards we need to move them in reverse order to not collide
+	if(going == UP)
+		var/obj/structure/industrial_lift/platform_to_move
+		var/current_index = length(lift_platforms)
+
+		while(current_index > 0)
+			platform_to_move = lift_platforms[current_index]
+			current_index--
+
+			platform_to_move.travel(going)
+
+	else if(going == DOWN)
+		for(var/obj/structure/industrial_lift/lift_platform as anything in lift_platforms)
+			lift_platform.travel(going)
+	set_controls(LIFT_PLATFORM_UNLOCKED)
+
+/**
+ * Moves the lift, this is what users invoke with their hand.
+ * This is a SAFE proc, ensuring every part of the lift moves SANELY.
+ * It also locks controls for the (miniscule) duration of the movement, so the elevator cannot be broken by spamming.
+ */
+/datum/lift_master/proc/MoveLiftHorizontal(going)
+	set_controls(LIFT_PLATFORM_LOCKED)
+
+	if(multitile_platform)
+		for(var/obj/structure/industrial_lift/platform_to_move as anything in lift_platforms)
+			platform_to_move.travel(going)
+
+		set_controls(LIFT_PLATFORM_UNLOCKED)
+		return
+
+	var/max_x = 0
+	var/max_y = 0
+	var/max_z = 0
+	var/min_x = world.maxx
+	var/min_y = world.maxy
+	var/min_z = world.maxz
+
+	for(var/obj/structure/industrial_lift/lift_platform as anything in lift_platforms)
+		max_z = max(max_z, lift_platform.z)
+		min_z = min(min_z, lift_platform.z)
+
+		min_x = min(min_x, lift_platform.x)
+		max_x = max(max_x, lift_platform.x)
+		//this assumes that all z levels have identical horizontal bounding boxes
+		//but if youre still using a non multitile tram platform at this point
+		//then its your own problem. it wont runtime it will jsut be slower than it needs to be if this assumption isnt
+		//the case
+
+		min_y = min(min_y, lift_platform.y)
+		max_y = max(max_y, lift_platform.y)
+
+	for(var/z in min_z to max_z)
+		//This must be safe way to border tile to tile move of bordered platforms, that excludes platform overlapping.
+		if( going & WEST )
```

Comment:
weird spacing on all these ``if(going & DIR)`` checks

Reply:
i think that was how it was before, gone now